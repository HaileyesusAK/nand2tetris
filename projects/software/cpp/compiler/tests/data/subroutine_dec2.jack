method boolean is_unit_circle() { return r = 1; }
