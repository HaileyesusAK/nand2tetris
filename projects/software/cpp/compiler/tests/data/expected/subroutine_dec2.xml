<subroutineDec>
    <keyword> method </keyword>
    <keyword> boolean </keyword>
    <identifier> is_unit_circle </identifier>
    <symbol> ( </symbol>
    <parameterList>
    </parameterList>
    <symbol> ) </symbol>
    <subroutineBody>
        <symbol> { </symbol>
        <statements>
            <returnStatement>
                <keyword> return </keyword>
                <expression>
                    <term>
                        <identifier> r </identifier>
                    </term>
                    <symbol> = </symbol>
                    <term>
                        <integerConstant> 1 </integerConstant>
                    </term>
                </expression>
                <symbol> ; </symbol>
            </returnStatement>
        </statements>
        <symbol> } </symbol>
    </subroutineBody>
</subroutineDec>
